src/App.tsx

```diff
@@ -2,50 +1,107 @@
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
+import cn from 'classnames/bind';
 import './App.scss';
+import { GoodsList } from './components/GoodsList/GoodsList';
 
 export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
+  { id: 1, name: 'Dumplings' },
+  { id: 2, name: 'Carrot' },
+  { id: 3, name: 'Eggs' },
+  { id: 4, name: 'Ice cream' },
+  { id: 5, name: 'Apple' },
+  { id: 6, name: 'Bread' },
+  { id: 7, name: 'Fish' },
+  { id: 8, name: 'Honey' },
+  { id: 9, name: 'Jam' },
+  { id: 10, name: 'Garlic' },
 ];
 
+interface Good {
+  id: number;
+  name: string;
+}
+
+enum SortBy {
+  abc = 'abc',
+  length = 'length',
+}
+
+function getReorderedGoods(
+  goods: Good[],
+  sortType: string,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType === SortBy.length) {
+    visibleGoods.sort((good1, good2) => good1.name.length - good2.name.length);
+  }
+
+  if (sortType === SortBy.abc) {
+    visibleGoods.sort((good1, good2) => good1.name.localeCompare(good2.name));
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<string>('');
```

Comment:
The initial state for `sortField` is an empty string, which does not match the `SortBy` enum values. Consider initializing it with a valid enum value or handling the empty state appropriately.